File: Python/lab15/Consultant.py
from ActiveSubscription import ActriveSubscription
from CallToClient import CallToClient
from CreateOrder import CreateOrder
from DeleteOrder import DeleteOrder


class Consultant:
    def __init__(self, call_to_client: CallToClient, create_order: CreateOrder, delete_order: DeleteOrder,
                 activate_subscription: ActriveSubscription):
        self.callToClient = call_to_client
        self.createOrder = create_order
        self.deleteOrder = delete_order
        self.activateSubscription = activate_subscription

    def create_order(self, name):
        self.createOrder.execute(name)

    def delete_order(self, name):
        self.deleteOrder.execute(name)

    def call_to_client(self, phone:str):
        self.callToClient.execute(phone)

    def activate_subscription(self, login):
        self.activateSubscription.execute(login)

File: Java/lab15/src/main/java/org/example/Consultant.java
package org.example;

public class Consultant {
    CallToClient callToClient;
    CreateOrder createOrder;
    DeleteOrder deleteOrder;
    ActivateSubscription activateSubscription;

    public Consultant(CallToClient callToClient, CreateOrder createOrder, DeleteOrder deleteOrder,
                      ActivateSubscription activateSubscription) {
        this.callToClient = callToClient;
        this.createOrder = createOrder;
        this.deleteOrder = deleteOrder;
        this.activateSubscription = activateSubscription;
    }
    public  void  createOrder(String name){
        createOrder.execute(name);
    }
    public  void  deleteOrder(String name){
        deleteOrder.execute(name);
    }
    public  void  callToClient(String phone){
        callToClient.execute(phone);
    }
    public  void  activateSubscription(String login){
        activateSubscription.execute(login);
    }
}
